internal/controllers/machineset/machineset_controller.go

```diff
@@ -1299,7 +1299,10 @@ func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, s *scope) (
 	for _, m := range machinesToRemediate {
 		log.Info("Deleting unhealthy Machine", "Machine", klog.KObj(m))
 		patch := client.MergeFrom(m.DeepCopy())
-		if err := r.Client.Delete(ctx, m); err != nil && !apierrors.IsNotFound(err) {
+		if err := r.Client.Delete(ctx, m); err != nil {
```

Comment:
Small bugfix that I took over from my code: don't go on with `conditions.MarkTrue`/`Patch` if the object doesn't exist